File: medium-and-easy/151-reverse-words-in-a-string/Solution151.js
/**
 * @param {string} s
 * @return {string}
 */
var reverseWords = function(s) {

    //NOTE, remember that in the fat arrow function we must return the result or else we get nothing back
    return s.trim().split(" ").filter( (str) => {return str !== ""} ).reverse().join(" ");

};

File: medium-and-easy/151-reverse-words-in-a-string/Solution151.java
import java.util.Arrays;

class Solution151 {
    public String reverseWords(String s) {

        String[] resultArray = Arrays.stream(s.trim().split(" ")).filter( str -> !str.equals("")).toArray(String[]::new);
        
        int n = resultArray.length;
        
        if (n > 1) {
            for (int i = 0; i < n/2; i++) {
                String temp = resultArray[i];
                resultArray[i] = resultArray[n-i-1];
                resultArray[n-1-i] = temp;
            }
        }
        
        return String.join(" ", resultArray);
        
    }
}